lib/regexp/regexp.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2021 The Bazel Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package regexp provides regular expression related functions.
+package regexp // import "go.starlark.net/lib/regexp"
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"sort"
+	"strings"
+
+	"go.starlark.net/starlark"
+	"go.starlark.net/starlarkstruct"
+)
+
+// Module regexp is a Starlark module of regular expression related functions.
+// The module defines the following functions:
+//
+//     compile(pattern) - Compile a regular expression pattern into a regular expression object, which
+//                        can be used for matching using its matches(), find(), find_all and other methods.
+//
+//     re.find(src) - Returns a string holding the text of the leftmost match in the given string of the regular expression re.
+//                    If there is no match, the return value is an empty string, but it will also be empty if the
+//                    regular expression successfully matches an empty string.
+//
+//     re.find_all(src, max) - Returns a tuple of all successive matches of the regular expression re. An empty tuple indicates
```

Comment:
Tuples are typically used like anonymous structs, where there are a fixed number of components identified by number. If the length can be arbitrary (including 0 or, confusingly, 1) a list is more appropriate.


"Returns a new, mutable list..."